lib/lcd.js

```diff
@@ -809,9 +822,23 @@ LCD.prototype.send = function(value) {
     pin++;
   }
 
-  ["LOW", "HIGH", "LOW"].forEach(function(val) {
-    this.io.digitalWrite(this.pins.en, this.io[val]);
-  }, this);
+  // At VCC = 3.3V, the minimum enable pulse width is specified as 450
+  // nanoseconds on page 49 of the HD44780 datasheet.
+  // https://www.sparkfun.com/datasheets/LCD/HD44780.pdf
+  // We therefore wait for 1 microsecond here to ensure that fast IO plugins
+  // like Pi-IO generate an enable pulse that's wide enough.
+  this.io.digitalWrite(this.pins.en, this.io.LOW);
+  this.io.digitalWrite(this.pins.en, this.io.HIGH);
+  sleepus(1);
+  this.io.digitalWrite(this.pins.en, this.io.LOW);
+
+  // The execution time for the vast majority of instructions is at least
+  // 37 microseconds. See datasheet pages 24 and 25.
+  // https://www.sparkfun.com/datasheets/LCD/HD44780.pdf
+  // It's important to wait 37 microseconds here to prevent fast IO plugins
+  // like Pi-IO from executing the next instruction before the current
+  // instruction has completed.
+  sleepus(37);
```

Comment:
If it's any help, I tested with a 2004 LCD on a Pi 3 and a 1602 LCD on an Arduino UNO and there were no problems. I'd say the `sleepus(1)` and `sleepus(37)` calls are low risk modifications. The modification to `sleep` is potentially riskier. `sleep` used to wait 1 millisecond longer than asked but now that the high resolution timer is being used it no longer waits 1 millisecond longer than asked.